cpp/server/ct-server.cc

```diff
@@ -317,17 +317,23 @@ int main(int argc, char* argv[]) {
   // Publish this node's hostname:port info
   cluster_controller.SetNodeHostPort(FLAGS_server, FLAGS_port);
 
-  // Periodically refresh our contiguous tree size so that other nodes get a
-  // good view of the data we have locally and can use us for replication.
-  // This has the side effect of refreshing the TTL on our ClusterNodeState
-  // entry.
+  // Periodically refresh our ClusterNodeState entry so it doesn't get deleted.
+  // TODO(alcutter): Figure out if we actually need this here now, the signer
+  // thread above will refresh the entry, but potentially not frequently
+  // enough.
   PeriodicClosure node_state_refresh(
       event_base, std::chrono::seconds(FLAGS_node_state_refresh_seconds),
       [&db, &cluster_controller, &internal_pool]() {
         // Actually run this on the thread pool, since this periodic closure is
         // calling back on the eventloop thread.
         internal_pool.Add([&db, &cluster_controller]() {
-          cluster_controller.ContiguousTreeSizeUpdated(db->TreeSize());
+          SignedTreeHead latest_sth;
+          if (db->LatestTreeHead(&latest_sth) !=
+              Database<LoggedCertificate>::LOOKUP_OK) {
```

Comment:
Dammit, yes.
Annoying because I was halfway through wiring in the `TreeSigner` here and then had an idea that just using the DB would be "easier" (not realising that it came with a free side of "completely wrong".)
I've re-done and finished off what I was going to do originally.